### PR TITLE
[13.0][FIX] base_rest: Includes default parameters and responses into the g…

### LINKED ISSUE #103: api-docs broken as headers are not documented with new implementation
Before the refactoring #71 every service was relying on `_get_openapi_default_parameters` to document endpoints (eg: required headers). 
When using `restapi.method` this is ignored and docs are not usable at all.

@lmignon shouldn't `RestApiMethodRoutePlugin` take care of this as well?

Example:

```
@restapi.method(
        [(["/", "/search"], "GET")], auth="api_key",
    )
    def search(self):
        return self._paginate_search()
```

IMO we should support old behavior and allow to pass per method parameters via decorator.

### PR DESCRIPTION
**Pull request by lmignon**
…enerated openapi doc

fixes #103

ping @simahawk

### CODE REVIEW

File: base_rest/apispec/rest_method_param_plugin.py [hunk 1]
```diff
@@ -31,30 +31,41 @@ def operation_helper(self, path=None, operations=None, **kwargs):
...
-                parameters.sort(key=lambda a: a["name"])
+            parameters = self._generate_pamareters(routing, method, params)
+            if parameters:
                 params["parameters"] = parameters
-            if output_param and isinstance(output_param, RestMethodParam):
-                responses = params.get("responses", {})
-                # get response from RequestMethodParam object
-                responses.update(self._default_responses.copy())
-                responses.update(output_param.to_openapi_responses(self._service))
+            responses = self._generate_response(routing, method, params)
```
Comment (simahawk) [inline base_rest/apispec/rest_method_param_plugin.py]:
```suggestion
            responses = self._generate_responses(routing, method, params)
```
or, rename var to `response`? Do you expect multiple items?

File: base_rest/tests/test_openapi_generator.py [hunk 2]
```diff
@@ -150,3 +150,58 @@ def update_name(self, _id, _name):
...
+
+            def _get_openapi_default_responses(self):
+                responses = super()._get_openapi_default_responses().copy()
+                responses["999"] = "TEST"
+                return responses
+
+        self._build_services(self, PartnerService)
+        service = self._get_service_component(self, "partner")
+        openapi = service.to_openapi()
+        self.assertTrue(openapi)
```
Comment (simahawk) [inline base_rest/tests/test_openapi_generator.py]: this assert seems superfluos

**Review by simahawk — APPROVED** [review]
:tada: merci!

File: base_rest/apispec/rest_method_param_plugin.py (hunk 1 above)
Comment (lmignon) [reply base_rest/apispec/rest_method_param_plugin.py]: responses is a list... I'll rename

File: base_rest/tests/test_openapi_generator.py (hunk 2 above)
Comment (lmignon) [reply base_rest/tests/test_openapi_generator.py]: :+1:

**Pull request merged**